<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="David Winter" />

<meta name="date" content="2022-10-06" />

<title>Connecting data to Open Tree trees</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap;
}
pre {

border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Connecting data to Open Tree trees</h1>
<h4 class="author">David Winter</h4>
<h4 class="date">2022-10-06</h4>



<div id="combining-data-from-otol-and-other-sources." class="section level2">
<h2>Combining data from OToL and other sources.</h2>
<p>One of the major goals of <code>rotl</code> is to help users combine data from other sources with the phylogenetic trees in the Open Tree database. This examples document describes some of the ways in which a user might connect data to trees from Open Tree.</p>
</div>
<div id="get-open-tree-ids-to-match-your-data." class="section level2">
<h2>Get Open Tree IDs to match your data.</h2>
<p>Let’s say you have a dataset where each row represents a measurement taken from one species, and your goal is to put these measurements in some phylogenetic context. Here’s a small example: the best estimate of the mutation rate for a set of unicellular Eukaryotes along with some other property of those species which might explain the mutation rate:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a>csv_path &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;protist_mutation_rates.csv&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;rotl&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a>mu &lt;-<span class="st"> </span><span class="kw">read.csv</span>(csv_path, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a>mu</span></code></pre></div>
<pre><code>##                     species       mu pop.size genome.size
## 1   Tetrahymena thermophila 7.61e-12 1.12e+08    1.04e+08
## 2    Paramecium tetraurelia 1.94e-11 1.24e+08    7.20e+07
## 3 Chlamydomonas reinhardtii 2.08e-10 1.00e+08    1.12e+08
## 4  Dictyostelium discoideum 2.90e-11 7.40e+06    3.40e+07
## 5  Saccharomyces cerevisiae 3.30e-10 1.00e+08    1.25e+08
## 6       Saccharomyces pombe 2.00e-10 1.00e+07    1.25e+08</code></pre>
<p>If we want to get a tree for these species we need to start by finding the unique ID for each of these species in the Open Tree database. We can use the Taxonomic Name Resolution Service (<code>tnrs</code>) functions to do this. Before we do that we should see if any of the taxonomic contexts, which can be used to narrow a search and avoid conflicts between different codes, apply to our group of species:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a><span class="kw">library</span>(rotl)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a><span class="kw">tnrs_contexts</span>()</span></code></pre></div>
<pre><code>## Possible contexts:
##    Animals 
##       Birds, Tetrapods, Mammals, Amphibians, Vertebrates 
##       Arthropods, Molluscs, Nematodes, Platyhelminthes, Annelids 
##       Cnidarians, Arachnids, Insects 
##    Fungi 
##       Basidiomycetes, Ascomycetes 
##    All life 
##    Bacteria 
##       SAR group, Archaea, Excavata, Amoebozoa, Centrohelida 
##       Haptophyta, Apusozoa, Diatoms, Ciliates, Forams 
##    Land plants 
##       Hornworts, Mosses, Liverworts, Vascular plants, Club mosses 
##       Ferns, Seed plants, Flowering plants, Monocots, Eudicots 
##       Rosids, Asterids, Asterales, Asteraceae, Aster 
##       Symphyotrichum, Campanulaceae, Lobelia</code></pre>
<p>Hmm, none of those groups contain all of our species. In this case we can search using the <code>All life</code> context and the function <code>tnrs_match_names</code>:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a>taxon_search &lt;-<span class="st"> </span><span class="kw">tnrs_match_names</span>(<span class="dt">names =</span> mu<span class="op">$</span>species, <span class="dt">context_name =</span> <span class="st">&quot;All life&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a>knitr<span class="op">::</span><span class="kw">kable</span>(taxon_search)</span></code></pre></div>
<table>
<colgroup>
<col width="23%"></col>
<col width="23%"></col>
<col width="16%"></col>
<col width="6%"></col>
<col width="10%"></col>
<col width="5%"></col>
<col width="13%"></col>
</colgroup>
<thead>
<tr class="header">
<th align="left">search_string</th>
<th align="left">unique_name</th>
<th align="left">approximate_match</th>
<th align="right">ott_id</th>
<th align="left">is_synonym</th>
<th align="left">flags</th>
<th align="right">number_matches</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">tetrahymena thermophila</td>
<td align="left">Tetrahymena thermophila</td>
<td align="left">FALSE</td>
<td align="right">180195</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">paramecium tetraurelia</td>
<td align="left">Paramecium tetraurelia</td>
<td align="left">FALSE</td>
<td align="right">568130</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">chlamydomonas reinhardtii</td>
<td align="left">Chlamydomonas reinhardtii</td>
<td align="left">FALSE</td>
<td align="right">33153</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">dictyostelium discoideum</td>
<td align="left">Dictyostelium discoideum</td>
<td align="left">FALSE</td>
<td align="right">160850</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">saccharomyces cerevisiae</td>
<td align="left">Saccharomyces cerevisiae</td>
<td align="left">FALSE</td>
<td align="right">356221</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">saccharomyces pombe</td>
<td align="left">Schizosaccharomyces pombe</td>
<td align="left">FALSE</td>
<td align="right">990004</td>
<td align="left">TRUE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p>Good, all of our species are known to Open Tree. Note, though, that one of the names is a synonym. <em>Saccharomyces pombe</em> is older name for what is now called <em>Schizosaccharomyces pombe</em>. As the name suggests, the Taxonomic Name Resolution Service is designed to deal with these problems (and similar ones like misspellings), but it is always a good idea to check the results of <code>tnrs_match_names</code> closely to ensure the results are what you expect.</p>
<p>In this case we have a good ID for each of our species so we can move on. Before we do that, let’s ensure we can match up our original data to the Open Tree names and IDs by adding them to our <code>data.frame</code>:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true"></a>mu<span class="op">$</span>ott_name &lt;-<span class="st"> </span><span class="kw">unique_name</span>(taxon_search)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true"></a>mu<span class="op">$</span>ott_id &lt;-<span class="st"> </span>taxon_search<span class="op">$</span>ott_id</span></code></pre></div>
</div>
<div id="find-a-tree-with-your-taxa" class="section level2">
<h2>Find a tree with your taxa</h2>
<p>Now let’s find a tree. There are two possible options here: we can search for published studies that include our taxa or we can use the ‘synthetic tree’ from Open Tree. We can try both approaches.</p>
<div id="published-trees" class="section level3">
<h3>Published trees</h3>
<p>Before we can search for published studies or trees, we should check out the list of properties we can use to perform such searches:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true"></a><span class="kw">studies_properties</span>()</span></code></pre></div>
<pre><code>## $tree_properties
##  [1] &quot;ot:nearestTaxonMRCAOttId&quot;         &quot;ot:branchLengthTimeUnit&quot;         
##  [3] &quot;xsi:type&quot;                         &quot;ot:MRCAOttId&quot;                    
##  [5] &quot;ot:specifiedRoot&quot;                 &quot;ot:nearestTaxonMRCAName&quot;         
##  [7] &quot;ot:nodeLabelMode&quot;                 &quot;edgeBySourceId&quot;                  
##  [9] &quot;ntips&quot;                            &quot;ot:MRCAName&quot;                     
## [11] &quot;tb:quality.tree&quot;                  &quot;tb:type.tree&quot;                    
## [13] &quot;ot:nodeLabelDescription&quot;          &quot;ot:ottId&quot;                        
## [15] &quot;ot:outGroupEdge&quot;                  &quot;ot:branchLengthMode&quot;             
## [17] &quot;ot:ottTaxonName&quot;                  &quot;ot:branchLengthDescription&quot;      
## [19] &quot;rootedge&quot;                         &quot;ot:nodeLabelTimeUnit&quot;            
## [21] &quot;ot:unrootedTree&quot;                  &quot;label&quot;                           
## [23] &quot;tb:ntax.tree&quot;                     &quot;ot:studyId&quot;                      
## [25] &quot;tb:kind.tree&quot;                     &quot;meta&quot;                            
## [27] &quot;ot:curatedType&quot;                   &quot;ot:reasonsToExcludeFromSynthesis&quot;
## [29] &quot;ot:rootNodeId&quot;                    &quot;ot:tag&quot;                          
## [31] &quot;ot:messages&quot;                      &quot;nodeById&quot;                        
## [33] &quot;ot:inGroupClade&quot;                 
## 
## $study_properties
##  [1] &quot;prism:section&quot;                 &quot;nexmljson&quot;                    
##  [3] &quot;ot:focalCladeOTTTaxonName&quot;     &quot;treesById&quot;                    
##  [5] &quot;prism:pageRange&quot;               &quot;ot:agents&quot;                    
##  [7] &quot;xhtml:license&quot;                 &quot;xmlns&quot;                        
##  [9] &quot;dc:title&quot;                      &quot;ot:focalClade&quot;                
## [11] &quot;ot:curatorName&quot;                &quot;prism:volume&quot;                 
## [13] &quot;prism:startingPage&quot;            &quot;ot:otusElementOrder&quot;          
## [15] &quot;treebaseId&quot;                    &quot;prism:creationDate&quot;           
## [17] &quot;tb:identifier.study&quot;           &quot;id&quot;                           
## [19] &quot;nexml2json&quot;                    &quot;ot:comment&quot;                   
## [21] &quot;skos:historyNote&quot;              &quot;ot:candidateTreeForSynthesis&quot; 
## [23] &quot;prism:publicationDate&quot;         &quot;ot:notIntendedForSynthesis&quot;   
## [25] &quot;prism:endingPage&quot;              &quot;tb:title.study&quot;               
## [27] &quot;dc:publisher&quot;                  &quot;dc:subject&quot;                   
## [29] &quot;ntrees&quot;                        &quot;dc:date&quot;                      
## [31] &quot;ot:studyYear&quot;                  &quot;dcterms:bibliographicCitation&quot;
## [33] &quot;ot:studyPublication&quot;           &quot;version&quot;                      
## [35] &quot;prism:doi&quot;                     &quot;skos:changeNote&quot;              
## [37] &quot;dc:creator&quot;                    &quot;ot:annotationEvents&quot;          
## [39] &quot;about&quot;                         &quot;prism:publicationName&quot;        
## [41] &quot;dc:contributor&quot;                &quot;ot:treesElementOrder&quot;         
## [43] &quot;generator&quot;                     &quot;prism:modificationDate&quot;       
## [45] &quot;tb:identifier.study.tb1&quot;       &quot;ot:taxonLinkPrefixes&quot;         
## [47] &quot;ot:studyPublicationReference&quot;  &quot;prism:number&quot;                 
## [49] &quot;ot:studyId&quot;                    &quot;ot:tag&quot;                       
## [51] &quot;ot:dataDeposit&quot;                &quot;ot:messages&quot;                  
## [53] &quot;otusById&quot;</code></pre>
<p>We have <code>ottIds</code> for our taxa, so let’s use those IDs to search for trees that contain them. Starting with our first species <em>Tetrahymena thermophila</em> we can use <code>studies_find_trees</code> to do this search.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true"></a><span class="kw">studies_find_trees</span>(<span class="dt">property =</span> <span class="st">&quot;ot:ottId&quot;</span>, <span class="dt">value =</span> <span class="kw">as.character</span>(<span class="kw">ott_id</span>(taxon_search)[<span class="dv">1</span>]))</span></code></pre></div>
<pre><code>##    study_ids n_trees                       tree_ids candidate study_year
## 1    ot_1587       1                          tree1                 2015
## 2    ot_1589       1                          tree1                 2015
## 3    ot_2037       4 tree10, tree11, tree12, tree13                 2018
## 4     ot_409       2                   tree1, tree2     tree2       2015
## 5     ot_564       1                        Tr85317   Tr85317       2015
## 6     ot_579       1                        Tr60046                 2013
## 7     ot_700       1                          tree1     tree1       2016
## 8      ot_73       1                          tree1     tree1       2013
## 9     ot_766       1                        Tr85440                 2015
## 10    ot_767       1                          tree1     tree1       2016
## 11     ot_87       1                        Tr64119   Tr64119       2014
##                                                                                                                                    title
## 1  &#39;Phylogenomic analyses reveal subclass Scuticociliatia as the sister group of subclass Hymenostomatia within class Oligohymenophorea&#39;
## 2  &#39;Phylogenomic analyses reveal subclass Scuticociliatia as the sister group of subclass Hymenostomatia within class Oligohymenophorea&#39;
## 3                                                                                                                                       
## 4                                                                         Tree of life reveals clock-like speciation and diversification
## 5     &#39;The alveolate translation initiation factor 4E family reveals a custom toolkit for translational control in core dinoflagellates&#39;
## 6                                       &#39;Convergent evolution of heat-inducibility during subfunctionalization of the Hsp70 gene family&#39;
## 7                                                                                                       &#39;A new view of the tree of life&#39;
## 8                                           Deep relationships of Rhizaria revealed by phylogenomics: A farewell to Haeckel’s Radiolaria
## 9                                                                                 &#39;Bacterial proteins pinpoint a single eukaryotic root&#39;
## 10                 &#39;Untangling the early diversification of eukaryotes: a phylogenomic study of the evolutionary origins of Centrohelida
## 11                        &#39;Dinoflagellate phylogeny revisited: Using ribosomal proteins to resolve deep branching dinoflagellate clades&#39;
##                                        study_doi n_matched_trees
## 1  http://dx.doi.org/10.1016/j.ympev.2015.05.007               1
## 2  http://dx.doi.org/10.1016/j.ympev.2015.05.007               1
## 3    http://dx.doi.org/10.1038/s41586-018-0708-8               3
## 4        http://dx.doi.org/10.1093/molbev/msv037               1
## 5    http://dx.doi.org/10.1186/s12862-015-0301-9               1
## 6      http://dx.doi.org/10.1186/1471-2148-13-49               1
## 7   http://dx.doi.org/10.1038/nmicrobiol.2016.48               1
## 8  http://dx.doi.org/10.1016/j.ympev.2012.12.011               1
## 9      http://dx.doi.org/10.1073/pnas.1420657112               1
## 10      http://dx.doi.org/10.1098/rspb.2015.2802               1
## 11 http://dx.doi.org/10.1016/j.ympev.2013.10.007               1
##            match_tree_ids
## 1                   tree1
## 2                   tree1
## 3  tree13, tree10, tree11
## 4                   tree2
## 5                 Tr85317
## 6                 Tr60046
## 7                   tree1
## 8                   tree1
## 9                 Tr85440
## 10                  tree1
## 11                Tr64119
##  [ reached &#39;max&#39; / getOption(&quot;max.print&quot;) -- omitted 16 rows ]</code></pre>
<p>Well… that’s not very promising. We can repeat that process for all of the IDs to see if the other species are better represented.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true"></a>hits &lt;-<span class="st"> </span><span class="kw">lapply</span>(mu<span class="op">$</span>ott_id, studies_find_trees, <span class="dt">property =</span> <span class="st">&quot;ot:ottId&quot;</span>, <span class="dt">detailed =</span> <span class="ot">FALSE</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true"></a><span class="kw">sapply</span>(hits, <span class="cf">function</span>(x) <span class="kw">sum</span>(x[[<span class="st">&quot;n_matched_trees&quot;</span>]]))</span></code></pre></div>
<pre><code>## [1]  51  51 129  71  17  88</code></pre>
<p>OK, most of our species are not in any of the published trees available. You can help fix this sort of problem by <a href="https://tree.opentreeoflife.org/curator">making sure you submit your published trees to Open Tree</a>.</p>
</div>
<div id="a-part-of-the-synthesis-tree" class="section level3">
<h3>A part of the synthesis tree</h3>
<p>Thankfully, we can still use the complete Tree of Life made from the combined results of all of the published trees and taxonomies that go into Open Tree. The function <code>tol_induced_subtree</code> will fetch a tree relating a set of IDs.</p>
<p>Using the default arguments you can get a tree object into your R session:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true"></a>ott_in_tree &lt;-<span class="st"> </span><span class="kw">ott_id</span>(taxon_search)[<span class="kw">is_in_tree</span>(<span class="kw">ott_id</span>(taxon_search))]</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true"></a>tr &lt;-<span class="st"> </span><span class="kw">tol_induced_subtree</span>(<span class="dt">ott_ids =</span> ott_in_tree)</span></code></pre></div>
<pre><code>## Warning in collapse_singles(tr, show_progress): Dropping singleton nodes with
## labels: mrcaott2ott276, mrcaott2ott142555, mrcaott2ott1551, mrcaott2ott7623,
## Chloroplastida ott361838, Chlorophyta ott979501, mrcaott185ott42071,
## mrcaott185ott1426, mrcaott1426ott1544, mrcaott1544ott8659, mrcaott1544ott15345,
## mrcaott1544ott9282, mrcaott9389ott818260, mrcaott9389ott23557,
## mrcaott23557ott527099, mrcaott148ott902, SAR ott5246039, Alveolata ott266751,
## Ciliophora (phylum in subkingdom SAR) ott302424, Intramacronucleata
## ott340382, mrcaott1546ott1671, Conthreep ott5248773, mrcaott1671ott16129,
## Peniculia ott1002116, Paramecium (genus in subkingdom SAR) ott568126,
## mrcaott11752ott13570, Hymenostomatia ott5257367, Tetrahymena (genus in
## subkingdom SAR) ott47284, mrcaott295406ott523463, mrcaott295406ott523462,
## Amoebozoa ott1064655, mrcaott3973ott15653, mrcaott3973ott26103,
## mrcaott26103ott273110, mrcaott26103ott229626, Dictyostelia ott835575,
## Dictyosteliales ott4008839, Dictyosteliaceae ott4008841, Dictyostelium
## ott999665, Opisthokonta ott332573, Nucletmycea ott5246132, Fungi ott352914,
## mrcaott109ott3465, mrcaott109ott67172, mrcaott109ott1423, mrcaott109ott9352,
## h2007-2 ott5576447, h2007-1 ott5584405, Dikarya ott656316, Saccharomycetales
## ott4085960, Saccharomycetaceae ott989994, Saccharomyces ott908546,
## Schizosaccharomycetes ott921286, Schizosaccharomycetidae ott5670481,
## Schizosaccharomycetales ott508517, Schizosaccharomycetaceae ott990009,
## Schizosaccharomyces ott990008</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true"></a><span class="kw">plot</span>(tr)</span></code></pre></div>
<div class="figure">
<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAfgAAAEgCAMAAABvm5EPAAAC7lBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxeXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29xcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn6AgICBgYGCgoKDg4OEhISGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////5XpBWAAAACXBIWXMAAAsSAAALEgHS3X78AAAUiklEQVR4nO2deXxVxdnHo4QlNEgChCQQIAlLBHnZ8oJhKVtEhUorBZWlQqUgBSoYlFoEWkHEqijUFrBIECwWQSxaDIvlFaGIKEsF1EAAwyLbmxDCzXaf/zpzzjNnuUsC5CYZfJ7vh2Quc2bmXO+P5AY/fv2FAUOSsJp+AkzNwMEThYMnCgdPFA6eKBw8UTh4onDwROHgicLBE4WDJwoHTxQOnigcPFE4eKJw8ETh4InCwROFgycKB08UDp4oHDxROHiicPBE4eCJwsEThYMnCgdPFA6eKBw8UTh4onDwROHgicLBE4WDJwoHTxQOnigcPFE4eKJw8ETh4InCwROFgycKB08UDp4oHDxROHiicPBE4eCJwsEThYMnCgdPFA6eKBw8UTh4onDwROHgicLBE0Xz4N/pHtl4ZhH+5k+TjWHBjOvbi8tDzYS/Bp7ftMj4yBsfn/wOqAH+1KHZGr8B4NCLVfLcbgC9g1/X8d/Fx5Pfwt+NW20MR3OvbzMut/AWh+Q5dTkSeH7yEuMj/SnP2rrX1LAsPe+ziCKfASCn+fSQPJdKoHfwvWR2S9fAzh9H9zsDd/0hNXIZlNT/JjExsV4GfNozqkc2wCtJceNLYPv/Nkj+AHChc7m6Ev9683O4o+kLdzXbObrZQFj5AMC2brj86ODGXfep48Qfr5S5Sa0+AvDdc63JU8ntvgC1HNSKWbU77BAfC9uKmSb7d5nDtbhjAM2/cQ8A3/ds+3YNvqoGegc/ouNbl8SwN/bj4rG/u3L7fO/yTnCgvZjZfNf57IStxU8Phc2dsy/Hf3m18XslLw7Eha7leCW71uxC3HEybAWMaZR9pcFhETr8+B/m8jNNNhQ93R2PE3dYU2dN6UvtwG/Pp2HLSpZ0BVwuwBXFkR75sXSU+HoOy8VhQz+xoP419wD5fT+Nya6pl1Shd/D5szuFpx+GEYvFV9jOf3UGyBoImaMBPmt7HCb9UYxt4OXe4ksIis/Cpd9MwIWu5XhlncgYd2zsCTB+PnijLn0TD1t64Kbf/Upk2AiPEzw5EuBcHf89rw4DuBhehsvBOnWfuJ34WJ+YnZ3a0IvDM3MBjrcA9+C5b1N24xp7SRUaBT95wkQXUwvl7MlRraDxGfnoj88ALMyA6Qvh6zbia7L5foBV98P5hyM6boKvhtw5oN1SXOhajldmzbV2/P5ZgG6fwzfJcC28tNdmXN4lQbyBDMbjBH3Fu8TeFP89vxB/TPakqOVgnbpsHMiPssdbDJ2YroaH1gIsH+YzjMiEtwfXxAvsQqPg0/9vr4uDkCDfRD9pXVCrFMoW5D2yDmDkahiQdebObQAlYQXii/CVgoPgWdixsIV4N+70ubnwtHM5XoEh76sdMOxdKIosgndGAMSs6A14euQJgG8vmMeJu5Y1EPd4YYL/npQtAPMngbkcrOcBv34N5Md3XvEPslIN3bPEe8nf3cPxqKio2hH31vCLrVPwPynwnRk+5kzx7i6rIWlD4awukCx+NGr3FURnd5V/T4KWb5T9PSl/T1RuUcZD+2Ou5i2s5cGFzuXqSvwptQMSj8G+rgDPzBdfxDEiXnP53bM9m2NOmceJ0w/f9svC3XHf+u3Ju2162Y7mp3G5/TwgbSvIj+h3SzL6Fanh50+VLeld5jMIev6zWl/ZQGgdfM6wH8X0F18m2xJif557MUq850eW5iQ8W6dNmzZzYVvbuKGnwDvhjoRHvi99IKbb621fMBe6luOVs/JN1dxxWVxYPh7g/g8BftYf8HQ40PmO3rvwODG5atiDUb12++/ZNvSnTdI+AVwO1qkwNvaC/FgdHzcpHwCHQ13iRub5DuInkohL1fnCBkTr4Ksab/cdwS5Nq/F/w1LFUA6+9C8/CXqtT9A/Ej8QKAffc9DFoNe2e6rxidQElIMnDQdPFA6eKBw8UTh4onDwROHgiRLC4PfsrRz9OPhqJITBt5pYOZKuhO65MBURwuD7V3I/f6uvTjh4onDwROHgicLBE4WDJwoHTxQOniiaB++SJl1UkRJpYGmZ5fiZ6EhenZLYaB7AuVphYWGLxfNNi360FAp/06zhNGsQPFN1z/Vm0Tt4tzTpIrRKpHu3pWWW42eiIzliatHhuqfg9akFBQWl8Lf+uRcSPoQpwy7mNNuiBjgxrV9lnlvVoHfwSpoMuRLpnpW7jYfPjQf4MvFq/avmwhLxAG8QxJF8P60AChrkQ79P5aUrCd8BnMy/csdZgMGZOMB7abFPV/KlqQL0Dh6lydArka5Zudt8+NZggAdWHmiPC8UDvEEwRxLg8rYxc+B03b5RvY/B6nueaD0oR+p6cCgmFwex5751lXxpqgC9g0dpMvRKpGtW7jYf7uwMuzqWZY7GheIB3iCYIwmwsWujg/CPUbklY8fClKiNl0aNEZNlK5pvtAdv9KlKvjRVgEbBd+uW6qK/JU2GXol0zcrd5sPcGBi4UWqZ5kLxAG8QzJGUzJhkDOsHwX2vAGwYCPBF6rDjYA1wNL6Sr0xVoFHw/qA0GXol0jUrd6ME6a33YS+AAVm4cECWukEQR/LcADE187lLy8UwdzZ0EO/0c38LG5M/ks8eB4CVD4b8lak8WgeP0mTolUjXrLEbH7ZL3gEQfR4XRp9XNwjiSOaHf+zd2uJM/u3ryz668wIM/3XRv5JPn4077PF4ynAQWybPD/krU3m0Dh6lydArka5ZYzdKkPeJvwnkJOBC8UDdOpgjuTg29t5DAH+Obzr0JMCxHo3Tj8DLYZLDOIgtqVtD/spUHq2DrxzlKJHMDzj48pRI5gccfHlKJPMDDp4pFw6eKBw8UTh4onDwROHgicLBE4WDJwoHTxQOnigcPFE4eKJw8ETh4InCwROFgyeK3sHvDwuvFXHPCYfEJpU1f+zZV6eVJ7ytHR7kKs6K6xWD1hygEmfqctglWDwrKXq2rdJZBYP6oXfwy9MBTgwZ7JDYpLLmjz37i1XlCW9PPB/kKs6K6xWD1hwqcajLYZfg7J8VfB3xrVLprIJBDdE7+Md/Kz59UbtISmyG7TardodfmUqcs25Qimwot6UcEWuL6xdCSaTH1ugkWXc3zEjdUqK8OLTnzGPkrHkdJbmAJ7isuR2mEoe6nOoSbPsZeOOPokpnFQzqiN7Bd3tXfMoJLzrQHm234kiPqcS56gbFLMpteVFesfaLjgAH/8eh0YlTdrX+qmDI7ZctL848D48Rs3gdzwl0gq81ZyhxqMupLsH2S67OS1cqnZrUEq2D99Q+KT5vv1NKbKbttq8zmEqcq25QzKLcttUoJHxjLMCqRx0anVhzfybAinZgeXHmeXiMmMXreE6gE3ytOUOJQ10OuwRhXp16DU8rlU5NaonWwe9uKj/PfExKbKbttmwcKnGuukExi3Lbgply7eTXADJedmh0IqUIkdKrY2wvzjwPj5m+UF3HcwKc4GfNGUoc6nLYJTjv4bzCkVOM575+kJrUk2oMvm3qjdG/cLH8L+PPNTsqbTbTdpPKmlTiXHWDUmRDuW34OrEW0j6BopbbnRodnKvrBRiyyPLizPMuogk3IEtdx3MCnODTLIhKHOpyZpdgaVPxp+n5CUqlw55BPanG4G/8O8Kjc6Ho47vmSJsNbTeprBlKnLNuUM6i3NbyhFwbsxmeDbvk1OjET2rvFS0K32l7ceZ5aMJFn1fX8ZwAJ/g0C6ISZ+py2CV4rd4m2Nn6kFLp7IJBDdE6+A7h9Rr0ecuw2dB2k8qaqcQ56gblrCm3fR9jrJ31o+4vJro0OoDMmC6LGhZaXpwS8Yxj5B7zupLkAp3gtuZQiTN1OdUl+HZy0377LJXOKhjUEa2DDwgrcSHhlguelbjQcMsFz0pcaLjlgmdCAwdPFA6eKBw8UTh4onDwROHgicLBE4WDJwoHTxQOnigcPFE4eKJw8ETh4InCwROFgyeK1sF7IsJrhYWHtzV/5/Ylb6xwUOy9/g24csEMqWAGOsrhTKIlaeuRcjJvfHzyO7Y6qSVaBy9Yl2o9NMxIqxnQt3CwfMTe628oxJVHc6WCGegohzOJlqTSI83J9Kc8a+teU+qknuge/MxJquxPioqyGRAbAI3CQewUtNVGVCebvJTa4E2rjRCMvdhQ6OgYtLeZOuTRlLlJrT7Cow2RMuWIfUZAZxItSaVHmpO75LeoJvtRndQU3YMf+IYq+yuO9MhmQGwANAsHTYHSVhtx6bGw57x/tdoIBWIvNhQ6OgbtbahDrqmzpvSldni0FCnzorz2GQGdSbQkbT1STi4dBZATlovqpKZoHry34QHlMe7rbDQDYgOgWThoCpS22ohL1yeVONoIwZAcsaHQ0TFob0Md8smR4n25jlqZOVoqmPYZAZ1JtCQtPdKYXJ+YnZ3a0IvqpKZoFPzhvW4Oirkj9UuVx7hsnNEMiA2AZuGgKVDaaiMunf2ss40QDMkRGwodHYP2NtQh+4p3gr0pauX0hVLBtM8I6EyiJWnpkcZk2eMthk5MN/6Z1g+6+dezatEo+C6PTXQxtVC80n1UF6AUFYe8D6oB0CwcNARKh9qIS4eud7YRgiE54gZHx6C1DXXIsgbbAF6YoFYOyILh6+wzAjqTaEnaeqQhUn7nFT/frVTqpKZoFHyg/vgnpltdgGlbZTOgagA0CwdNgdJWG3FpwnFnGyEYkiNucHQM2ttMHfLwbb8s3B33rVoZfR5annCcEciZREvS1iMNkTL63ZKMfkVKndQUzYPvtcbqAhwb+5/GoBoAsXDQECgdaqO59IKrjVAeMzb2a9xgdww6tpk65KphD0b12g14dE6CVDAdZwR0JtGStPRIY3J1fNykfEud1JQQBt9qYvkkVbA/UPAVEFKBctqLoTtLf0IY/J695XN3BftvPPjQCpR9SFm4IQy+Im7mW335hFag3O4J4WHac0sHz9w8HDxROHiicPBE4eCJwsEThYMnCgdPlBoMvo/P/8I05kr1PRemBoP3/T1/xVcnHDxROHiicPBE4eCJwsEThYMnCgdPFA6eKBw8UfQO/j9h4eHhHXZf/z2C9MrahnXgblqbIN206EYr8dnsk1WKtOlGX52S2GgeaN0n60Tv4N8cAHBphOM/pQ3uNpsE6ZW1u2fdrrX/uUG6adGNRvEZ+2RRkUY3esTUosN1T2ndJ+tE7+CnyG7ZOaOVq+ywpP3MaKMqNkivrN0963at1Vo54+ymDepGm+Iz9smqIlnTjf7/tAIoaJCvdZ+sE42C7/naUhdvFkOPDVC2q9kH6Co7LGk/M9qsig3SK2t3z7pda1wrZ1zdtMHcaBSfsU8WFWlAYRoubxszR+8+WScaBf/GX3yDL65bp16dlsuVq+ywpP3MaLMqNkivrN0963atca2ccXXTBnOjUXzGPllUpJUwDRu7Njqod5+sE42C92dvvDmiq+ywpP3MaLMqNnCvrKN71u1a41o54+qmDe5GG+Iz9slaRbJH8WnCjEl698k60Tr4Pz9gDMpVdljSvmY0Vs8G7pV1dM+6XWtcK51pZzdtMDdaic9mn6xSpE03+pz4MRRmPqd3n6wTrYN/bI4xKFfZYUn7mdFmVWzgXllH96zbtca10pl2dtMGc6OV+Gz2ySpF2nSj88M/9m5tcUbvPlknWgffaYMxoKt81rak/cxosyo2WK+s1T3rdq1xrTwXnN20Qd1oFJ+xTxYVaXSjF8fG3ntI8z5ZJ1oHz1Qd1Rh8mo823aP6bs34UY3Bj/H5HyW0rL5bM35UY/C+8Lf6moSDJwoHTxQOnigcPFE4eKJw8ETh4InCwROFgycKB08UDp4oHDxROHiicPBE0Tz4K0+n1O970Cp/DNT8aLN2eCWejS1OBa6kFIdvWiRXuQwrVKtQn1IWlelUoW9lT2qF3sFf6fTkqdMzWpeq8sdAzY82Tzx/40/CcqdscSpwJaU4fPISucplWKFahfoUDuhUoW+lJjVD7+BHZ4hPxZ2PY/mjo/nRMqSwbDLr7oYZqVvQeALfDkmf6slXkuLGl8jzbXdKilO4CO/l0LbMw2fV7rCt/lW5EHzUqvdNfQotKnSqsGhSTeqG1sGfiEKdDssf7eZH25AyyyZ3tf6qYMjtl9F48uuQdFdPbu6cfTn+S3Gsw5060F4twns5tC08vDjSI1aJX+BfO2noU9ZgOFVW0aSa1Autg18wEh9g+aPd/GgbUmbZ5P2ZACvaKePJr0PSXT35cm+UGh3uVOZotQjv5dC28PB9neUq8Qv8aycNfcoaDKfKKprESc2oweBTKmitmlo4VsqysO8DVf5oNz/ahpRRNumNEK/0q2OU8eTXIemunjz/cETHTfJkhzs1faFahPeytS11+LJxcpX4BYHUqhmT7MFyqrBoEq9pRQ0Gf6iC1qqDMEf2NpZ0Wq3KH+3mR4chJcsmz9X1AgxZpIwn3w5Jd/VkwUHwLOwon4HDnRqQhYvwXg5tCw+XHpVYJX75qlWoTymLynSq0LeyJzWjBoOvmH/X/9u1A/c8DKr80W5+dBhSsmyyJPK9okXhO5Xx5Nsh6a6e3BOVW5TxkDzf4U5Fn8dFeC+HtqUOT9sqV4lf4KtWmfqUsqjQqTJ9K3tSM7QOHja1bdB9sVeVP9rNj05DSpZNQmZMl0UNC9F48uuQdFdPeifckfDI92KTw53KSVCLLrrWGkvw8LGxnyfIhRK3WoX6FA7oVKFvZU1qht7BXwchLZskxK0efGjLJglxqwcf2rJJQtzqwTM3CQdPFA6eKBw8UTh4onDwROHgicLBE4WDJwoHTxQOnigcPFE4eKJw8ETh4InCwROFgycKB08UDp4oHDxROHiicPBE4eCJwsEThYMnCgdPFA6eKBw8UTh4onDwROHgicLBE4WDJwoHTxQOnigcPFE4eKJw8ETh4InCwROFgycKB08UDp4oHDxROHiicPBE4eCJwsEThYMnCgdPFA6eKBw8UTh4onDwROHgicLBE4WDJwoHTxQOnigcPFE4eKJw8ETh4InCwROFgycKB08UDp4oHDxROHiicPBE4eCJwsEThYMnyn8B6HYB3QDTmOUAAAAASUVORK5CYII=" alt />
<p class="caption">plot of chunk subtree</p>
</div>
</div>
<div id="connect-your-data-to-the-tips-of-your-tree" class="section level3">
<h3>Connect your data to the tips of your tree</h3>
<p>Now we have a tree for of our species, how can we use the tree and the data together?</p>
<p>The package <code>phylobase</code> provide an object class called <code>phylo4d</code>, which is designed to represent a phylogeny and data associated with its tips. In oder to get our tree and data into one of these objects we have to make sure the labels in the tree and in our data match exactly. That’s not quite the case at the moment (tree labels have underscores and IDs appended):</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true"></a>mu<span class="op">$</span>ott_name[<span class="dv">1</span>]</span></code></pre></div>
<pre><code>## $`Tetrahymena thermophila`
## [1] &quot;Tetrahymena thermophila&quot;</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true"></a>tr<span class="op">$</span>tip.label[<span class="dv">4</span>]</span></code></pre></div>
<pre><code>## [1] &quot;Dictyostelium_discoideum_ott160850&quot;</code></pre>
<p><code>rotl</code> provides a convienence function <code>strip_ott_ids</code> to deal with these.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true"></a>tr<span class="op">$</span>tip.label &lt;-<span class="st"> </span><span class="kw">strip_ott_ids</span>(tr<span class="op">$</span>tip.label, <span class="dt">remove_underscores =</span> <span class="ot">TRUE</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true"></a>tr<span class="op">$</span>tip.label <span class="op">%in%</span><span class="st"> </span>mu<span class="op">$</span>ott_name</span></code></pre></div>
<pre><code>## [1] TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<p>Ok, now the tips are together we can make a new dataset. The <code>phylo4d()</code> functions matches tip labels to the row names of a <code>data.frame</code>, so let’s make a new dataset that contains just the relevant data and has row names to match the tree</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true"></a><span class="kw">library</span>(phylobase)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true"></a>mu_numeric &lt;-<span class="st"> </span>mu[, <span class="kw">c</span>(<span class="st">&quot;mu&quot;</span>, <span class="st">&quot;pop.size&quot;</span>, <span class="st">&quot;genome.size&quot;</span>)]</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true"></a><span class="kw">rownames</span>(mu_numeric) &lt;-<span class="st"> </span>mu<span class="op">$</span>ott_name</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true"></a>tree_data &lt;-<span class="st"> </span><span class="kw">phylo4d</span>(tr, mu_numeric)</span></code></pre></div>
<p>And now we can plot the data and the tree together</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true"></a><span class="kw">plot</span>(tree_data)</span></code></pre></div>
<div class="figure">
<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAfgAAAFoCAMAAACMkBkOAAAC+lBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwuLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///8KYwt0AAAACXBIWXMAAAsSAAALEgHS3X78AAAcV0lEQVR4nO2de1wVZf7H2fasWVumibeW3ajdLi5lqyWXg4KCeE/FvFuaKLqWlWlW6k8lF7fA1EzNyjRFLdSUzE0zsxITXVSyg6Ch4gUJRVRQ4XA5z+v1e2bmmTPfuZxzcDwcmJ3v+4/DnO9zmWnfzrnMmX0+fgQxJX71fQBI/YDiTQqKNyko3qSg+Jvk8FsjY/qOX1lc38dxq6D4m+Kr8PH/OWu/mv1B99Fn6vtYbg0UfxOUD5/kPNPTrRvr81BuGRRfe25EbYDPhi2ttyPxAijeSeHCEb3GfmZ33WHYJtnT6qd31vER1SUoXiQ5cv3ZspykkB9dddg8WVG4Elxex8dUh6B4xtQZDv5vScw32h0c1hJl6cMFGv1s3Zs/MET8KGAL4v9kdKjlUbD+3iZlhKqE4gV2PC9ulYZc0eyR8bKqVBmp7lZz//ySirnta4RnTOTV/bU8DJX4iloOdA+Kd0m3C87Nz/6t2WPGLnVtsPo73bnflVL7Ay+S9X9tFV9ua/v6/cG7SGYHsi4gIOC2NLLhIf8BF0l1vL//TEIWBd7dMZcIPWX9WUtWxPT2bITtia73hn4TfF8iGU1fZqZPZoO+f7z1oEvidFRwbPsm1uNENSalX+y9wTbCuvOgeJ7LvaTtG1GgoWLnVbZFXap4S+PjXVD4Ru4l42irnKLgZJvfYsesME48ZfOjN/KaHbCPG0Y2PFFyqsmx/Ma/VIyPZz1l/VlL1h/fqWQjbH6bb4QEXDpy5/VN0XQf6UL34qZ7qqf0YdPRPaT4bayc056oxqT4baqc83cH687jRfHLJk7TzWnvHcbNcjWPsu1FUInIk5jwane2FVyjHrsiWX0Frywx+PbwPWQ2/SiYvdf2hzLyc5Ag/lybwyR5DCFFtzt2B/5IZ7MXkUsvDmE9Zf1ZS1azGnGErbWDvPESIX/OL737Wl7rGqH7CvqP4NxtpcJ0lJSnCKlqdko1Rqj/yrrzXb0oftoHmbqpx4/HVaWUQ+NAJbxU4pOwmWyri8YXvSWfan77u5JsORW3iNuyteXftznxNRHzCZkyhxYbFzvmPtjsRXvVq39+MnIIEXrK+rOWrLbOEbZH6JvNXEIC80l0WvJ41n22fxDlvDAdfZ7yDH14PF01htVZd/4YvSh+9nfem8vX2DtL2wWxsOWguPFsvnrYm+mq0qqe3KN19fRphGRu4D6sMfFzo+jXhuQ4Qoob1ZwpJDntUtf9o5isHkKEnrL+rCUryDkCiF8Ub93JBi0bS//h7iXCdHS3KR0JqfbPU40R6vmsOw+KFxh9wLk5G16fI4fEjcVr1KO6lqpKhc2WXq3a5n/q4H15xdb5kviM1tyZdvzew1UTBpGk6IKSDqsWd6m8Yh1AhJ6y/qyFE89GAPF5Lfyr2KAzLffZZ0Sy6Qj3Hp9Wk/iYQzUmxe/LmsSOhHXnQfEC+eFlbOtgtAM2OMWfGaAaVNhXY6ZjPVo2DaMf+lY84D/aLokfeVdgYOAy8vlDLfpfIKV9m7ScUHUlqpV1besvhJ6y/qyFE89GAPHkkecIm55sfbRpVD6bjhZTYno3D8lRj0np1al55xOEdedB8YwdEXn8361h52V1p3gy8BBRMNHFtZ46xprmqkXj+7orULzIL72fWbRmbsSkq/KyZDu3a6W8KbMXqQeqDjV1+VEYxevizMZZX5cpi+A0XzFe9iZQEFxY98ekZnObj1y25e+r9TQoHuDIUtfg6/u/ht+QnhwM1uhtHFA8wJN4sjHkc3YZ5+IrPc6rOxsIFA/wKJ6Uzgkds2DdyoQePbb56JjqChQP8CyecjQtYflO1S+0hgPFA2olnpDjqk+ABgTFA1C8LlC8kUDxABSvCxRvJFA8AMXrAsUbCRQPQPG6QPFGAsUDULwuULyRQPEAFK8LFG8klOKHBUfp5U/v18t/gRcxs/hpGbqnwjPeSKB4AIrXBYo3EigegOJ1geKNBIoHoHhdoHgjgeIBKF4XKN5IoHgAitcFijcSKB6A4nWB4o0Eihep2TdvUp8pS04qykrxv614o//YhJ2KNRKMB4oXsC8KfmXbL6cOrBvadYesQS7+YJ9+H+89mf3tzLAZipUzjAaK5zkW9r54DhdNHHpdLBcunxQzccEJ8Wn11Gfy2KYjteMeXx6g10HxHBlheeDZls7CMsbnx/T65Mj57NTB/XP551Wx74Fel7rLlkUzGiieUhBaJHu+s1c1fcwMFpeut3X6kvszaZmsl71bbdekboigeEo/5Se4eVRxXuhvzufXu9N/A9+NUvQqDK2q80OrM1A8IXvilJWKkOuOrsdB4VLwdRKpWvQm6WON2WRBBfKWukkh4HFGIdQ2EwHFEzIqR1Wav+HLN2SFVUlHxql6XeuqnkweVCDDyykEstHOKITaZiKgeFITpq7ljRgi/0JfEfGWxnJH/X5TlZxBBV5PIZBV+dH8ZteV9K1paGYH1pFbL5ntGsQSqEHxpGCYRjGik6LQK1YtWfM/mQUVeD+FQFblRgub78cSEpxGfQsd6QbbAYwlUONF8RMnL9eNxv+ovuL6dmW6FEdn5av4+PbV6l4fLVbn17CgAu+nEMiq3GhhM6eJvfBeO/UtdKQbbAcwlkCNF8XvWZ2qG+38H59QvkudMkTFd1EUxj6p8RH+g+XXtKbkggq8n0Igq3Kj2eaTOz98nvMtdKQbbAcwlkCNF8UblaJBGsWIcEUhZvA5da//+0FVEoMKvJ9CIKtyo9lm4ss9d3LihY5c7pGwAxhLoAbFE0eoQ1X7ZeyYbFmhLCpZI0u2p3qhQzGowPspBLIqN5ptHv3LX6o58UJHusF2AGMJ1KB4QiYcVJUSvvp+gqzw7vJfh6t6lXTXmI0FFXg/hUBW5Uez3IGHX+Df2oWOdEPcNYglUIPiCclSvdaXhlSSfvAL8emwKtLzV2W36XVwtd51CoF3QfGUZ79VFCZ/Rt/6Q3Kdz4vCbfSltq/iLeFYpPo94hZxl0LgXVA8pST0hOz52tHc4wlrCrv+tiOEP/vnzZL1uhyqegm4ZdylEHgXFM9xLBi+rr/fV7gaem1O6OytB75OinyRhUxOnAK+y5+wqj/SGwgUz1PUY6r4Ad329GtOvRXbk0fN+0K6xW5xpPhLbXmSNZcYGRTP2NIldunW9E2JXYfIv8fJf7E9F99p5md7vl7xrPUjA/8ky4HinZxblxT33lfKS3HKn+ordy19Yfanx4nRQfEAvL3apKB4k4LiTQqKNyko3qSgeJOC4k0KijcpKN6koHiTguJNCoo3KSjepKB4k4LiTQqKNyko3qSgeJOC4k0KijcpKN6koHiTguJNCoo3KSjepKB4k4LiTQqKNykq8WVfJI9KXHdRUUXx/2soxJ8eFZX01YGvl/Tqf0QqHpnVq2OnoSsv+/jQvA6KB8jFr+v0E9vK6T2PrXZzcsDI7dfoP4kPrO9oLHRpJFA8QCZ+ybPS8tCOWZP4v3tCfmaV6gV9jf2Cj+IBUPzup2Wn9EvcokQ5VrCgYVqs1tLkhgHFA4D4aqt8we+q0BLiiJAtZf6WrxaoqhNQPACI3zBX0UYLm2bKKuUhWuFzXS0Wv99bLGzh256w6SYTKnrexADWM6MDyWpXuxEoHgDED1EGD1ZEkIFn5KWZ32hPE5ju3LTwQ9mTmxRvuYlIC9bz6n4UrwcgXplTQEjMDeWC1nve1J6GEy+kQ/TxCzrARUkIkRFCQoUQQiGFUrCutrBhLYO/c6ZLEG7sIRZpIYVSgGFCuEVKbPsm1uNsam5FW068cw43GF78oRX6V8nn4xHKi5ykheaxrd/Uqz4/nx6jqJweVFSktV49FS+mQ1j4KAkWGSEkVAghFFIoBetq81voSLCK6RIcFjaAgFAKaRgLt0jx21g5p73YUxAP5nCN4cXPGas/F4Nf2vl6gZOUv9vY1lm1+LG7eigqZ2MLCrQSFqh4MR3CwgdRsMgIIaFCCKGQQilYV1ujMpLtjK/gsLABBIRSSMNYuEXKU/SDZ7NTrKcgHszhGsOLX/K59+YCL/WdVY09rylf/n+apj0NFS+mQ1j4IAoWGcESKvgQCmmlctbV9jAhuc74Cg6LOACEUkjDWLhFyjP0z+PprKcgHszhGhQPAOJjzyraKsNJP0XMR4JGLhUHFS+mQ1j4heVZZARbr54PoZAMsq5cIdcZX8FhEQeAUAppGAu3SOlIv3r657Gegngwh2tQPACIX/Ouom3bDLJ2nqxSFeJihXEqXkyHsFRwulhkBLPDh1BIBllXJp515bBUsAEglEIaxsItUvzSahIfc8jEgzlcg+IBQHxliDwm3NG5kFSHFcLSwgUupuE+1bN0iH4BB6gSFhnB7PAhFCCUQujKxIvpEoQbe4gNkEIpwDAh3CIlpnfzkBwiEw/mcA2KB8BLtmkjZE0JSfQhs8t1qfJDd63rN7XBmyEUKSM899EExQNkP9LMfhVci186nP957qsIZ8bLmq4uQxzd490QChTvDeQ/yy7sfYxtFY56iS1SfiRqMpdcdO2LmKl6g2G9G0KRv0/nQBQPUNyIkdV34KeZ+T+nPtdll1T8dkLniPCYtxVXb40Higeobr06u3xyr5cWKq9+4q1XDYE6FU/wZssGC4rXB4oHoHgDgeL1geIBZhYftrZeDkM/KF4feMYDULyBQPH6QPEAFG8gULw+UDwAxRsIFK8PFA9A8QYCxeujYYif82h7vbSZ6r3DQPG+Jmmr7qF4xusDxQNQvK9B8T4HxQNQvK9B8T4HxQNQvK9B8T4HxQNQvK9B8T4HxQNQvK9B8T4HxQNQvK9B8T4HxQNU4qt+Wv/6mt03FFUU7zUapviTcdbXlqV+PDNikPy0R/FeoyGKr5kVs59tHh8cBxexQPFeowGKt8cmg5YNkSCfBMV7jQYo/vk1sqb0btJSR/uKvLfLegPFA+A6d1MUbR/NEP7ahloH9ApNvE4MDooHSOLtYcrViR1d+LUuV3XLpo/Vn1rzibFB8QBJ/OdJqsYvE+jDNwPY8lfZVuWXPBH/31ssgctrv9OMDtr1nhpbmmS1czmHG1A8QBI/VJlTwCcVEEeYM3bs42RVDwH/TFKx9rZsMND9Tq/u165b5FvqWcQKFe9qDjegeIAkvrNGa1QlyXjJ+cyujjIQoOIJeWA9SwvIAjkFymwCsv6vreLLMzuQdCs98a2wA59TYNdKO2B9+XlZgEE7binTWqUTAAwvPnlepm5KuQkqS51ssZ5kWxEaexp5tPQ9sGxE1xKuo/pEpOIrU393mKUFyHIKFNkER1vlFAUnA/HODtw8Fu20A1E8rYgBBrz4WqUTAAwvPnXcNN0c5Sa4mufkzYe3CRu/qnMKCInblTdro/S0ezbXs1jVzb9RY0ur98S0AJhToMwmmD2ZflbYC8Q7O3DzWLTTDkTxtCIGGPDia5VOADC8eG/icK4PqnXG97hGtv1LehrmYhL+pd4ZTgBzCpTZBHGLuI5M/D4rXJqacOI10w5YX64iBhjw4muVTgBA8QDpPb6HOjvWEUrIjTDnwsY/vuBiEiaepQXAnAJlNsH0aVT7Bk48nXqlWrxm2gHry1XEAANefK3SCQAoHiCJX7RO1ZjJmX5H/Jp3PVwZYSHCxLO0AJhToMwmOHhfXrF1PpV25I4z5Z1U4is00w5YX64iBhjw4muVTgBA8QBJfGEXh7LxOe4rU82wefwX+bPRX7mahIlnaQFZIKdAmU1AVjzgP9pOpTmmBtw9QSm+X4BdK+2A9eXmFQMMePG1SicAoHgAuGT7xmeKtoMDhS7vhkxfvezZqEwfHladgOIBQHyZ9ais6WKI+NJu/27eJ8d9eFB1BIoHwF/nTgXDuy9Oh/8kPcGfZb1GAxRPCiKTxaszNZ8G/wJaULzXaIjiSdXy4Ne+L6gszEgInSv7rQ7Fe40GKZ6Q6h/eHBE85NX/KK7Loniv0UDFc+Dt1R4Z91SUXgISbu2/wkugeF3M/k73UDzjfQ6KB6B4XaB4I4HiASheFyjeSKB4AIrXBYo3EigegOJ1geKNBIoHoHhdoHgjgeIBKF4XKN5IoHgAitcFijcSKB6A4nWB4o0EigegeF2geCOB4gEoXhco3kigeACK1wWKNxIoHoDidYHijQSKB6B4XZhEvD1n7e4CXxxN3YLiAWrxhbve3i5brnjf4M7xs6cOCZtZ4rvDqhNQPEApPq1r/zkfJA7rtKKaFcqGP3eC36jeEqZeF8tQoHgn9h1vjV5wWHpeFvuqcFrf+HeX8/xGcfg2qfeYhvH/8NULihdZHzJrx/4v4qLERU8qoiTLB0Mu0MfKmHQ4IG61D4/O66B4xpzxwgL0pzp/LxT+CRc828etGf/eQtkIu/UiMS4oXmBznLh1OZSPnMmWrxI55UtSEVwpH/Of1zSn+rjdHx98mxBbEP8sq5273V7x13GsIlI8AduXxuQ9uV5aOQYonscRUurc3vky9zh5r6xDUV+ybY5yUJhq+UvKu3/fX3Hg0SV1KF5ckkeKJ1CJrxAnt3C9tHIMUDzP/kngSSi3UnF4jbxHpP1V1f80zx9Tz3Tpnhz6uH0QsbV9/f7gXbx4ZVDB6AWETJ9MFgQE/pu62fCQ/4CLMKWAFcKGtQz+ThxcHe/vP5ObHyQUZHYQO7F9wXAEYfI+fkE/deBzDJQ0DPHzPjqhl1PVnqd3S80NylIYNjbkJK0oFy4flTPwN+XQhG03VHvf+RjbsPktdswK48Srggo2RRMSlL67dXZZN3+S1+yAfdww0OosLHQkWMXBG54oOdWE+3cGEgo48UInti8QjsAmp2d8ZkMWv3jQON1ke57eLZdzKW/C1X/jduTmZkcpuk3cGqG6Uvvu0lxVBN2HMWzD9ocy8nMQJ14VVFB697W81jUvzCIk3Z8kj6FvJLc7pFax0KiMZAeJg3cH/ii8BIGEAk680IntC4QjsMkbuvh6ZyP8wB7DLWfYWdFjQHFcnnLU1APqmbbzZ/y1Dyptbfn3XipeFVRAotOSx5OBqwgp8CdTuI8OjYulVrHwMCG5QeJgx9wHm71opw0goYATL3Ri+wLhCGxyFO+BSzHSdkk099jngqxDTRhZoIo96n5FPdOFu7jVjzf/hf/AJYhXBRWQRfHWneTF2YRspWc8/T5R3KhGagUF6pQNPlNIctql0plBQgEnXujE9gXCEdjkKN4TcdLBv7KFe/xkkaz969fIicGKMRd7aM2U2Ha/PeP+ZCBeFVRA8lr4V5Hv2xwr792CHL/3cNWEQaAVFKhTNjgpuqCkAz2LYUKBSjwIR2CTE0sFindLcUgO21o9kv9jD4U/w1Ra6Zt5P5t8zMvbiAaORY/e+bd3aoB4dVABeeQ52nN+QOCyQEI+f6hF/wvwn4VUoE7Z4NK+TVpO4DISQEKBSjwIRxAn7xfQoD/VNwBOhr/PXbo7F/88+5r8Q0+7s9ERv5I+Zne1wxE/9da/N2ua/rHeQSl+YlB7vbR4v17+C7xF5ZLIyN7W/t86C+u7FbKt0uF8BiBJGQ7M20J0X7GtOtS03HOvukUp/hYw9hnPUSO/Wre/8xvcj3W/vh0qXl9aHS1mUzjWhJ/WvZ/NbT7SPdZboHiA8vd4x/b4yM6RIzdI5/mR3s9+Wegoz32v0yyDB4mjeEBtbr2yJQ7pGD1mrTqXzmCgeICZb7a8BVC8kUDxABSvCxRvJFA8AMXrAsUbCRQPQPG6QPFGAsUDULwuULyRQPEAFK8LFG8kUDwAxesCxRsJFA9A8bpA8UYCxQNQvC5QvJFA8QAUrwsUbyRQPADF6wLFGwkUD0DxukDxRgLFA1C8LlC8kUDxABSvCxRvJFA8AMXrAsUbCRQPQPG6QPFGAsUDULwuULyRQPEAFK8LFM/T9PeWO4J3wxiBntodneWsdppRAowr/i5aWVVn1gGKB9RK/LWE0L5fu5mkqY2UpdxzCKQDWLQ7OstUvFaUAIOK1W5lVRR/69RGfE3v9ZUXhqa6nqQpt/Dp9MH8OqLr/9oqvryPX5CdRRMIEQRC6ABXFgIHqHjaOd1KT2IrDCygCHEDtJUFFfATsmm4HbCsA2EezRlcgeIBc5/8VlkqGa1YrDzrn/ShvIvrSXjxP/6N83K0VU5RcDJ3agvRBCyCgIUOWAgLHJCLdwYW0GlY3ABtFcYIE7JpaJW1s3m0ZnCJF8W/Hhqlm1zvHcbNcumwk9DUIYcVLG77irywKJkbFCE+VYeN8uJz7+K8zJ5MSPZeTrwQTcAiCFjogEUMHJCLdwYW0C4sboBTzI8RJmTT0CprZ/NozeASL4o3Pj+8flZZqv7+hrxwui99yO/rehJe/B7+jI8Tlj7n3sz5aAIWQcBCByxi4IAkfp8VrmLtjBugrcIYYUI2Da2ydjaP1gwuQfE3y1sjt64KcZOHwoufyb/HT59GSOYGXjwfTcAiCFjogEUMHGDiQwlZqdDG4gZoqzBGmJBNQ6usnc2jNYNLUPxN89+kDy+5aabib6Q2PcTZOnhfXrF1PrFUECGagEUQsNABSwULHBDEH7njTHknhTYWN0BbhTHChGwaWmXtbB6tGVyC4r1N00aN7+S+x3Mfulc84D/aTvoF2Fk0gRBBwEIHaFkIHBDEO6YG3D1BqU2IG6CtbIwwoTANtwMWRyDMozmDK1C8j6j/aAI5KN4nNIRoAjko3ic0hGgCOSjepKB4k4LiTQqKNyko3qSgeJOC4k0KijcpKN6koHiTguJNCoo3KSjepKB4k4LiTQqKNyko3qSgeJOC4k0KijcpKN6koHiTguJNCoo3KSjepKB4k4LiTQqKNyko3qSgeJOC4k0KijcpKN6koHiTguJNCoo3KSjepKB4k4LiTQqKNyko3qSgeJOC4k0KijcpKN6koHiTguJNCoo3KSjepKB4k4LiPVB6vjYl44HiJawCsHSpz21BeyPzPZSMCIqXyMjI2LcyfAIsPTOlIKhqerSHkhFB8QrKm8BnrapJEKlo5qFkRFC8gjMB8Nlje6jlvW09lIwIipfg3uBD75gKS7vv6eY/ttVWDyUjguIl0jmOyGsXV8xedsZjyYCgeMD1fA5YSePT45d5KBkRFC+ReOefAymwdNujh+mjp5IRQfESbY6oSgFftJnvkFvWKBkRFC/R8ZKqFEDORkSfD/RQMiIoXmJHwEtzKbBEv9vVzG19l4eSEUHxEiGhb8ygwNI67mHfSA8lI4LiJZqV1/cR+BAUL9Fli0Ne6LC/g4DbkjFB8RJP+LV+hCIVMq5mCrgtGRMUL2ETkNWOEXvqqipPJQOC4t0yowl5oV2Hf3ooGREU75bmp2qa5V5q6aFkRFC8W5pf/uFBUnCPh5IRQfFuiX8iMLEw+mkPJSOC4t1SvWG340xSmYeSEUHxnrm8vTYlg4HiPbO3cW1KBgPF3zx4X70ZOJmeJ7+Qi/fVm4Fj/2j2uP/jubCE99WbgbCEKlL9r1BYwvvqzUAT7pp8tT8s4X31ZqD9JvqwpQ8s4X31ZiDW76lnnvKLGTFin1TD++pNQIoI+BSvvvveiKB4DziKa+QFjbvvjQiKd8uFobc3bjToAixp3H1vRFC8W/rHF5HiCQNgSePueyOC4t3S3E4fKlvAksbd90YExbvlb/+lD4cegiWNu++NCIp3S2qzMXPi7v0clv5H7r5H8e75NXlK8nFZRXX3vTFB8R5Q/TqnuvvemKB4t2j8Oqd1970B+X+ZWlXEzDmPNQAAAABJRU5ErkJggg==" alt />
<p class="caption">plot of chunk unnamed-chunk-1</p>
</div>
</div>
</div>
<div id="find-external-data-associated-with-studies-trees-and-taxa-from-open-tree" class="section level2">
<h2>Find external data associated with studies, trees and taxa from Open Tree</h2>
<p>In the above example we looked for a tree that related species in another dataset. Now we will go the other way, and try to find data associated with Open Tree records in other databases.</p>
<div id="get-external-data-from-a-study" class="section level3">
<h3>Get external data from a study</h3>
<p>Let’s imagine you were interested in extending or reproducing the results of a published study. If that study is included in Open Tree you can find it via <code>studies_find_studies</code> or <code>studies_find_trees</code> and retrieve the published trees with <code>get_study</code>. <code>rotl</code> will also help you find external. The function <code>study_external_IDs</code> retrieves the DOI for a given study, and uses that to gather some more data:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true"></a>extra_data &lt;-<span class="st"> </span><span class="kw">try</span>(<span class="kw">study_external_IDs</span>(<span class="st">&quot;pg_1980&quot;</span>), <span class="dt">silent =</span> <span class="ot">TRUE</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">inherits</span>(extra_data, <span class="st">&quot;try-error&quot;</span>)) {</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true"></a>  extra_data</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true"></a>}</span></code></pre></div>
<pre><code>## External data identifiers for study 
##  $doi:  10.1016/j.ympev.2006.04.016 
##  $pubmed_id:  16762568 
##  $nucleotide_ids: vector of 58 IDs
##  $external_data_url http://purl.org/phylo/treebase/phylows/study/TB2:S1575</code></pre>
<p>Here the returned object contains an <code>external_data_url</code> (in this case a link to the study in Treebase), a pubmed ID for the paper and a vector IDs for the NCBI’s nuleotide database. The packages <code>treebase</code> and <code>rentrez</code> provide functions to make use of these IDs within R.</p>
<p>As an example, let’s use <code>rentrez</code> to download the first two DNA seqences and print them.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true"></a><span class="kw">library</span>(rentrez)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true"></a>seqs &lt;-<span class="st"> </span><span class="kw">try</span>(<span class="kw">entrez_fetch</span>(<span class="dt">db =</span> <span class="st">&quot;nucleotide&quot;</span>, <span class="dt">id =</span> extra_data<span class="op">$</span>nucleotide_ids[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], <span class="dt">rettype =</span> <span class="st">&quot;fasta&quot;</span>), <span class="dt">silent =</span> <span class="ot">TRUE</span>)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true"></a><span class="cf">if</span> (<span class="kw">inherits</span>(seqs, <span class="st">&quot;try-error&quot;</span>)) {</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true"></a>  <span class="kw">cat</span>(<span class="st">&quot;NCBI temporarily down.&quot;</span>)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true"></a>} <span class="cf">else</span> {</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true"></a>  <span class="kw">cat</span>(seqs)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true"></a>}</span></code></pre></div>
<pre><code>## &gt;AM181011.1 Plectroninia neocaledoniense partial 28S rRNA gene, specimen voucher G316300 (Queensland Museum)
## GCTAGCAAGCGCGTCGGTGGTTCAGCCGGCTGGTCTCGTCGAGTTGTCGGTGTGCGGATCCGAACGGACC
## GCGGCCGATGGCGTCGGCGGGCAAGCTGTGGTGCACTCTGTCGGCGTGCGCGTCAGCGTCGGTTTCGGCC
## GGACGACGAGGCGCTCGGGGAAGGTAGCTGGACCGGTCTTCGGTGCAGTGTTATAGCCCTGGGCCGCTGG
## GTTCGGCGTTTGGGACCGAGGAGAGAGATGATCGCTGCAGCGCCTGTCTCCCTCTCGAGGGGGGCTAGCC
## AGCCGCTGTTTGGGTGGCGTCACTGGCGGAGGACTGCACGCAGTGCTTCGCCGGTGGTCGTGTCCAGGCG
## GGCGGTGTGGGTATAGAGGCGCTTAGGACGCTGGCGTCCAAATGGCCGTGCGCGACCCGTCTTGAAACAC
## GGACCAAGGAGTCTAGCATGTGCGCGAGTCTTAGGGTGTGGAAGCCCTCGGGCGCAATGAAAGTGAAGGG
## CCGTCGTCTCTCGGGGCTGCGGTGTGAGGTGAGAGCCGTCGCCGTCGGGTGGCGGTGCATCATCGGCCGG
## TCCATCCTGCTCTCAGGAGGATCTGCGCAAGAGCGTGTTTGCTGGGACCCGAAAGATGGTGAACTATGCC
## TGAATAGGGTGAAGCCAGAGGAAACTCTGGTGGAGGCTCGTAGCGGTTCTGACGTGCAAATCGATCGTCA
## AATTTGGGTATAGGGGCGAAAGACTAATCGAACCATCTAGTAGCTGGTTCCCTCCGAAGTTTCCCTCAGG
## ATAGCTGGAACTCGTCTTGACACAGTTTTATCAGGTAAAGCGAATGATTAGAGGTCTTGGGGGTGAAACA
## CCCTCAACCTATTCTCAAACTTTAAATAGGTAAGAAGCGCGACTTGCTCAATTGAAGTGGCGCGCAGTGA
## ATGTGAGTTCCAAGTGGGCCATTTTTGGTAAGCAGAACTGGCGATGCGGGATGAACCGAACGCTCGGTTA
## AGGTGCCCAAGTCGACGCTCATCAGACCCCAGAAAAGGTGTTGGTCGATATAGACAGCAGGACGGTGGCC
## ATGGAAGTCGGAATCCGCTAAGGAGTGTGTAACAACTCACCTGCCGAATCAACTAGCCCTGAAAATGGAT
## GGCGCTCAAGCGTCGCACCTATACCGAGCCGTCGTGGTAAATGCCAGGCCACGACGAGTAGGAGGGCGCG
## GTGGTCGTGACGCAGCCCTTGGCGCGAGCCTGGGCGAAACGGCCTCCGGTGCAGATCTTGGTGGTAGTAG
## CAAATATTCAAATGAGAGCTTTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTGGACATGG
## GTTAGTCGATCCTAAGAGATAGGGAAGTTCCGTGTGAAAGTGCGCAATGCGCTTCTGTGCTGCGCGCCTC
## CTATCGAAAGGGAATCGGGTTAATATTCCCGAACCGGAAGGCGGATATCTCTGGCTCTCGGGTCAGGAGC
## GGCAACGCAAGCGTACTGCGAGACGTCGGCGGGGGCTCCGGGAAGAGTTGTCTTTTCTTTTTAACGCAGT
## CGCCATCCCTGGAATCGGTTTGCCCGGAGATAGGGTTGGCTGGCTCGGTAAAGCAGCACACTTCATGTGC
## TGTCCGGTGCGCTCTCGACGGCCCTTGAAAATCGCAGGTGTGCATCGATTCTCGCATCCGGTCGTACTCA
## TAACCGCATCAGGTCTCCAAGGT
## 
## &gt;AM181010.1 Eilhardia schulzei partial 28S rRNA gene, specimen voucher G316071 (Queensland Museum)
## GCTAGTAATGTACGTTGGTGGTTCAGCCGGCTAGTCTTGTCGAGTCGTCGTGTGGTGGATCCGACTGGAC
## CGTCCGCGGTGGTGTCGGCGGGCGAGCTGTGGTGCACTCTACGGACGTGCGCGTCAGCGTCGGTTCTCGA
## TGGGCGATAAGGTGCGTGGGGGAAGGTGGCTCGGTCCTTGGGAACTGAGTGTTACAGACCCTGGTGCTGG
## GCTCGTCGTGGGACCGAGGAGAGAGAGAGATGATCGCTGCGGCACCTGCCCCGTTGTCATTTTTCGGGGC
## TAGCCAGCCGTTTGTCAGGTGTGCGTCGGACGTTGAGGACTGCACGCAGTGCTGGACGTGGAGGCGTGAT
## CTGATGGCGGTGTGGGCATTAGAGGTGCCTAGGACGCTGGCGTCCAAATGGCCGTGCGCGACCCGTCTTG
## AAACACGGACCAAGGAGTCTAACATGTGCGCGAGTCTTAGGGTGTGCAAGCCCTCGGGCGCAATGAAAGT
## GAAGGCTCGGCGGCGCTAGTCGAGCTGAGGTGAGAGCCGTGGCCGTTGCATGTGGCGGCGGCGGCGCATC
## ATCGGCCGGTCCATCCTGCTCTCAGGGGGATCCGAGCAAGAGCGTATTTGTTGGGACCCGAAAGATGGTG
## AACTATGCCTGAATAGGGTGAAGCCAGAGGAAACTCTGGTGGAGGCTCGTAGCGATTCTGACGTGCAAAT
## CGATCGTCAAATTTGGGTATAGGGGCGAAAGACTAATCGAACCATCTAGTAGCTGGTTCCCTCCGAAGTT
## TCCCTCAGGATAGCTGGAGCTCTTGGACACAGTTTTATCAGGTAAAGCGAATGATCAGAGGTCTTGGGGG
## TGAAACACCCTCAACCTATTCTCAAACTTTAAATCGGTAAGAAGCGCGACTTGCTGAATTGAAGCCGCGC
## GCAAGCAATGTGAGTTCCAAGTGGGCCATTTTTGGTAAGCAGAACTGGCGATGCGGGATGAACCGAACGC
## TGGGTTAAGGTGCCAAAGTCGACGCTCATCAGACCCCAGAAAAGGTGTTGGTTGATATAGACAGCAGGAC
## GATGGCCATGGAAGTCGGAATCCGCTAAGGAGTGTGTAACAACTCACCTGCCGAATCAACTAGCCCTGAA
## AATGGATGGCGCTCAAGCGTCGCACCTATACCGGGCCGTCGTCGCAAATGCCAGGCGACGACGAGTAGGA
## GGGCGCAGTGGTCGTCATGCAGCCCTTGGCGTGAGCCTGGGTCAAACGGCCTCTGGTGCAGATCTTGGTG
## GTAGTAGCAAATATTCAAATGAGAGCTTTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTG
## GACATGGGTTAGTCGATCCTAAGTGATAGGGGAGCTCCGTATGAAAGTGCGCAATCGGCCCTGCTTGTGT
## CGCCTTGCGCCACCTATCGAAAGGGAATCGGGTTAATATTCCCGAACCGGAAGGCGGATTTTCTCTGGCT
## CTCGGGTCAGGAGCGGCAACGCTAGCGAACCGCGAGACGTCGGCGGGGGCTCCGGGAAGAGTTGTCTTTT
## CTTTTTAACGCAGTCGCCATCCCTGGAATCGGTTTGCCCGGAGATAGGGTTGGCTGGCTCGGTAAAGCAG
## CACACTTCATGTGCTGTCCGGTGCGCTCTCGACGGCCCTTGAAAATCGCGGCGAGTGTAGTCTGATTTTC
## GCATCCGTTCGTACTCATAACCGCATCAGGTCTCCAAGGT</code></pre>
<p>You could further process these sequences in R with the function <code>read.dna</code> from <code>ape</code> or save them to disk by specifying a file name with <code>cat</code>.</p>
</div>
<div id="find-a-ott-taxon-in-another-taxonomic-database" class="section level3">
<h3>Find a OTT taxon in another taxonomic database</h3>
<p>It is also possible map an Open Tree taxon to a record in another taxonomic database. For instance, if we wanted to search for data about one of the tips of the sub-tree we fetched in the example above we could do so using <code>taxon_external_IDs</code>:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true"></a>Tt_ids &lt;-<span class="st"> </span><span class="kw">taxon_external_IDs</span>(mu<span class="op">$</span>ott_id[<span class="dv">2</span>])</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true"></a>Tt_ids</span></code></pre></div>
<pre><code>##   source       id
## 1  silva AY102613
## 2   ncbi     5888
## 3   gbif  7415807</code></pre>
<p>A user could then use <code>rgbif</code> to find locality records using the gbif ID or <code>rentrez</code> to get genetic or bibliometric data about from the NCBI’s databases.</p>
</div>
</div>
<div id="what-next" class="section level2">
<h2>What next</h2>
<p>The demonstration gets you to the point of visualizing your data in a phylogenetic context. But there’s a lot more you do with this sort of data in R. For instance, you could use packages like <code>ape</code>, <code>caper</code>, <code>phytools</code> and <code>mcmcGLMM</code> to perform phylogenetic comparative analyses of your data. You could gather more data on your species using packages that connect to trait databases like <code>rfishbase</code>, <code>AntWeb</code> or <code>rnpn</code> which provides data from the US National Phenology Network. You could also use <code>rentrez</code> to find genetic data for each of your species, and use that data to generate branch lengths for the phylogeny.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
